pkg/daemon/certificate_writer.go

```diff
@@ -109,13 +111,32 @@ func (dn *Daemon) syncControllerConfigHandler(key string) error {
 			return err
 		}
 
+		for _, CA := range controllerConfig.Spec.ImageRegistryBundleData {
+			if err := os.MkdirAll(filepath.Join("/etc/docker/certs.d", CA.File), defaultFilePermissions); err != nil {
+				return err
+			}
+			if err := writeFileAtomicallyWithDefaults(filepath.Join("/etc/docker/certs.d", key, "ca.crt"), CA.Data); err != nil {
```

Comment:
Hmm, so both bundledata and bundleuserdata write to the same directory and named `ca.crt` ? Is there any possibility of conflict between the two?

Reply:
if ca.file was the same, then yes. But I am assuming that we want users to be able to overwrite the CA if they really want to? @flavianmissi WDYT?